browser/mac/sparkle_glue.mm

```diff
@@ -190,9 +190,16 @@ - (void)registerWithSparkle {
   // Background update check interval.
   constexpr NSTimeInterval kBraveUpdateCheckIntervalInSec = 3 * 60 * 60;
   [su_updater_ setUpdateCheckInterval:kBraveUpdateCheckIntervalInSec];
-  [su_updater_ setAutomaticallyChecksForUpdates:YES];
+
   [su_updater_ setAutomaticallyDownloadsUpdates:YES];
 
+  // We only want to perform automatic update checks if we have write
+  // access to the installation directory. Such access can be checked
+  // with SUSystemUpdateInfo:systemAllowsAutomaticUpdatesForHost.
+  // The following makes su_updater_ call this method for us because
+  // we setAutomaticallyDownloadUpdates:YES above.
+  if ([su_updater_ automaticallyDownloadsUpdates])
```

Comment:
IMO, we also don't need to downalod implicitely when we don't have writes. WDYT?
Or this can affect explicit update check request?
```
if ([su_updater_ automaticallyDownloadsUpdates]) {
  // Background update check interval.
  constexpr NSTimeInterval kBraveUpdateCheckIntervalInSec = 3 * 60 * 60;
  [su_updater_ setUpdateCheckInterval:kBraveUpdateCheckIntervalInSec];
  [su_updater_ setAutomaticallyDownloadsUpdates:YES];
  [su_updater_ setAutomaticallyChecksForUpdates:YES];
}
```

Reply:
Thank you for reviewing and the feedback @simonhong. We do need to download implicitly for two reasons:

1. For explicit ("on-demand") updates via `brave://settings/help`.
2. As explained in the comment above the snippet you highlighted, `setAutomaticallyDownloadUpdates:YES` changes the behavior of `automaticallyDownloadsUpdates` in a way we want, namely by making `automaticallyDownloadsUpdates` return the same value as `systemAllowsAutomaticUpdatesForHost`. See [here](https://github.com/brave/Sparkle/blob/07933da3e178265d0f0ba86e02bbde38e701a04d/Sparkle/SUUpdater.m#L482).

Reply:
@mherrmann Got it. Thanks for explanation!